core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/UniqueIDUserStoreManager.java

```diff
@@ -423,16 +423,72 @@ void updateUserListOfRoleWithID(String roleName, String[] deletedUserIDs, String
 
     // Group centric methods........................................................................................
 
+    /**
+     * Get group using the given group name.
+     *
+     * @param groupName       Group name.
+     * @param requestedClaims Claims required as claim URIs.
+     * @return Group Object with details.
+     * @throws UserStoreException If an error occurs while retrieving a group.
+     */
+    default Group getGroupByGroupName(String groupName, List<String> requestedClaims) throws UserStoreException {
+
+        return null;
+    }
+
+    /**
+     * Get the id of the group which matches the given group name.
+     *
+     * @param groupName Group name.
+     * @return Id of the group which matches the given group name.
+     * @throws UserStoreException If an error occurred while getting the group id.
+     */
+    default String getGroupIdByGroupName(String groupName) throws UserStoreException {
+
+        return null;
+    }
+
+    /**
+     * Get the name of the group which matches the given group id.
+     *
+     * @param groupId Group id.
+     * @return Name of the group which matches the given group id.
+     * @throws UserStoreException If an error occurred while getting the group name.
+     */
+    default String getGroupNameByGroupId(String groupId) throws UserStoreException {
+
+        return null;
+    }
+
     /**
      * Get group details using group ID.
      *
-     * @param groupID            Group ID.
+     * @param groupID         Group ID.
      * @param requestedClaims Claims required as claim URIs.
      * @return Group Object with details.
      * @throws UserStoreException If an error occurs while retrieving a group.
      */
     Group getGroup(String groupID, List<String> requestedClaims) throws UserStoreException;
 
+    /**
+     * Retrieves list of groups evaluating the condition.
+     *
+     * @param condition Conditional filter.
+     * @param limit     Number of search results. If the given value is greater than the system configured max limit
+     *                  it will be reset to the system configured max limit.
+     * @param offset    Start index of the user search.
+     * @param domain    Userstore domain.
+     * @param sortBy    Sorted by.
+     * @param sortOrder Sorted order.
+     * @return List of Group objects.
+     * @throws UserStoreException If an error occurs while listing groups.
+     */
+    default List<Group> listGroups(Condition condition, int limit, int offset, String domain, String sortBy,
```

Comment:
can't we pass the domain in the condition?

Reply:
We should not pass the domain in the condition. The only way we can pass the domain is in the attribute value(Ex: `secondary/*somi*). In this case we can extract the domain using the domain separator `/`. But attribute value can take any value (Eg: *2021/12*). In this if we extract the domain we get `2021` which is actually the value we want to search. Passing domain with the condition leads to additional overhead to the UMs. We are already following this approach [1].

[1] - https://github.com/wso2/carbon-kernel/blob/4.7.x/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java#L9871